internal/fs/buffer/write_buffer.go

```diff
@@ -17,25 +17,29 @@ package buffer
 const (
 	// MiB is the multiplication factor to convert MiB to bytes.
 	MiB = 1024 * 1024
-	// InMemoryBufferThresholdMB is the upper limit on the size upto which the buffer should
-	// be created in memory. Beyond this size, buffer should be on disk.
-	InMemoryBufferThresholdMB = 50
+	// maxInMemoryBufferSizeMB is the upper limit on the size upto which the
+	// buffer would be created in memory. Beyond this size, buffer would be on disk.
+	MaxInMemoryBufferSizeMB       = 50
+	NonSequentialWriteError       = "non-sequential writes are not supported with WriteBuffer"
+	NotEnoughSpaceInCurrentBuffer = "not enough space in currentBuffer to write entire content"
+	ZeroSizeBufferError           = "buffer of size 0 cannot be created"
 )
 
-// ChunkSize (bytes) is the size of data to be written in 1 write call to GCS.
-// Ensure ChunkSize <= 100MB to avoid memory bloat.
-var ChunkSize int
-
 // WriteBuffer is an interface that buffers the data to be written to GCS during
 // the write flow.
 // WriteBuffer is used only in create new file flow with sequential writes and
 // at any point in time, only 2x of the configured buffer size is stored in the
 // write buffer.
+// Sample usage:
+// // Create in-memory write or on-disk write buffer
+// writeBuffer, err := buffer.CreateInMemoryWriteBuffer(bufferSize)
+// // check err
+//
+//	for n times {
+//		err = buffer.WriteAt([]byte("content"), offset)
+//		// check err
+//	}
 type WriteBuffer interface {
-	// InitializeBuffer assigns a buffer of 2*configured buffer size to the WriteBuffer.
```

Comment:
@gargnitingoogle suggested that instead of relying on the ensureWriteBuffer, we should lazily allocate memory to buffers during the write calls itself. So, I removed this from method from ensureWriteBuffer and WriteBuffer interface and added ensureCurrentBuffer and ensureFlushedBuffer helpers to WriteAt method's logic.